Add missing page-level `applies_to` for ingest docs

vishaangelova opened the pull request:
## Summary

This PR adds/updates page-level `applies_to` information for:

- [x] docs in the `manage-data/ingest` folder
- [x] `reference/apm-agents/index.md` (+ adds some links)
- [x] docs in the `reference/fleet` folder:
  - [x] "Fleet and Elastic Agent overview" page (`index.md`)
  - [x] "Beats and Elastic Agent capabilities" page
  - [x] "Migrate from Beats to Elastic Agent" section
  - [x] "Deployment models" section
  - [x] "Install Elastic Agents" section
  - [x] "Secure connections" section
  - [x] "Manage Elastic Agents in Fleet" section
  - [x] "Configure standalone Elastic Agents" section
  - [x] "Manage integrations" section
  - [x] "Command reference” page
  - [x] “Agent providers” section
  - [x] "Agent processors" section
- [x] `reference/ingestion-tools/index.md` (+ adds some links)

## Generative AI disclosure

1. Did you use a generative AI (GenAI) tool to assist in creating this contribution?
- [x] Yes  
- [ ] No

2. Tool(s) and model(s) used:

Cursor / gpt-5.1 model

## Review assignments

| Commit Hash | Commit Title | Reviewer | Done |
|------------|--------------|----------|------|
| [33ab065](https://github.com/elastic/docs-content/commit/33ab06590a246b2733fc4d9271ae812fde54e052) | Add applies_to to reference/apm-agents/index.md | @theletterf | ✅ |
| [875e40c](https://github.com/elastic/docs-content/commit/875e40cc2077f042f91a700f34c98786438928aa) | Add applies_to to reference/ingest-tools/index.md | @bmorelli25 | ✅ |
| [93c0d87](https://github.com/elastic/docs-content/commit/93c0d879562ecd68b29952f99e3ed72846c75555) | Add applies_to to manage-data/ingest docs | @karenzone | ✅  |
| [ff9a870](https://github.com/elastic/docs-content/commit/ff9a870e481ae5f9efe28e1de84029f75af914a2) | Add applies_to to agent processor docs | @alexandra5000 | ✅ |
| [1e4576b](https://github.com/elastic/docs-content/commit/1e4576bb3155a439faaa0c1034f06ae89d202c71) | Add applies_to to more pages | @bmorelli25 | ✅ |
| [2c6519f](https://github.com/elastic/docs-content/commit/2c6519fdd92434d1b25a5049b3355ac1ee526e3b) | Add applies_to to 'Manage integrations' docs | @bmorelli25 | ✅ |
| [70857ac](https://github.com/elastic/docs-content/commit/70857ace4813e3ebc2a662a5c1460bfd42538517) | Add applies_to to 'Deployment models' docs | @bmorelli25 | ✅  |
| [a008244](https://github.com/elastic/docs-content/commit/a008244cc55203e875ff2770dac822b9be934430) | Add applies_to to 'Secure connections' docs | @bmorelli25 | ✅ |
| [1178589](https://github.com/elastic/docs-content/commit/1178589aeeba4cbc78d81c86edde1bc7420714a6) | Add applies_to to fleet/index.md | @bmorelli25 | ✅ |
| [5adeb75](https://github.com/elastic/docs-content/commit/5adeb75e3f8087374f29d8bd58a797dafe6ae1b4) | Add applies_to to 'Install Elastic Agents' docs | @karenzone | ✅  |
| [b6e14c6](https://github.com/elastic/docs-content/commit/b6e14c610cea3f5d6535a1de559e7fea017428b3) | Add applies_to to 'Manage Elastic Agents in Fleet' docs | @karenzone | ✅  |
| [7483e33](https://github.com/elastic/docs-content/commit/7483e33d83b4552933132ab5149965413ccfd332) | Update applies_to of 'Agent providers' docs | @karenzone | ✅ |
| [a0d4edb](https://github.com/elastic/docs-content/commit/a0d4edbd83607008dae7e48f0a3db240fb2e41e9) | Add applies_to to 'Configure standalone Elastic Agents' docs | @karenzone | ✅  |

vishaangelova commented on reference/fleet/migrate-auditbeat-to-agent.md:
```diff
@@ -24,20 +27,20 @@ The following table describes the integrations you can use instead of {{auditbea
-| [Osquery](integration-docs://reference/osquery/index.md) or[Osquery Manager](integration-docs://reference/osquery_manager/index.md) integration | Collect data from the [process](https://www.osquery.io/schema/5.1.0/#process) table on some OSeswithout polling. |
-| [System.socket](beats://reference/auditbeat/auditbeat-dataset-system-socket.md) dataset | [Endpoint](/solutions/security/configure-elastic-defend/install-elastic-defend.md) | Best replacement because it supports monitoring network connections on Linux,Windows, and MacOS. Includes process and user metadata. Currently does notdo flow accounting (byte and packet counts) or domain name enrichment ( ...
-| [Osquery](integration-docs://reference/osquery/index.md) or [Osquery Manager](integration-docs://reference/osquery_manager/index.md) integration | Monitor socket events via the [socket_events](https://www.osquery.io/schema/5.1.0/#socket_events) tablefor Linux and MacOS. |
+| [Osquery](integration-docs://reference/osquery/index.md) or[Osquery Manager](integration-docs://reference/osquery_manager/index.md) integration | Collect data from the [process](https://www.osquery.io/schema/5.1.0/#process) table on some operation systems without polling. |
```
```suggestion
| [Osquery](integration-docs://reference/osquery/index.md) or[Osquery Manager](integration-docs://reference/osquery_manager/index.md) integration | Collect data from the [process](https://www.osquery.io/schema/5.1.0/#process) table on some operating systems without polling. |
```

vishaangelova commented on manage-data/ingest/ingest-reference-architectures/agent-ls-airgapped.md:
```diff
@@ -1,6 +1,10 @@
 mapped_pages:
   - https://www.elastic.co/guide/en/ingest/current/agent-ls-airgapped.html
+applies_to:
+  stack: ga
```
```suggestion
```

vishaangelova commented on manage-data/ingest/ingest-reference-architectures/airgapped-env.md:
```diff
@@ -1,6 +1,10 @@
 mapped_pages:
   - https://www.elastic.co/guide/en/ingest/current/airgapped-env.html
+applies_to:
+  stack: ga
```
```suggestion
```

vishaangelova commented on reference/fleet/deployment-models.md:
```diff
@@ -1,6 +1,13 @@
+    ece: ga
+    eck: ga
+    self: ga
+    ess: ga
```
```suggestion
  stack: ga
```

bmorelli25 commented:
Thanks for doing this, @vishaangelova.

@theletterf, @colleenmcginnis, @karenzone, and @alexandra5000 — I've assigned you all to various aspects of this PR (see description). The goal is to review your sections and determine if the applicability badges look correct. If you need clarification from the dev/product team on any pages/features, please call them out.

bmorelli25 commented on reference/fleet/fleet-server.md:
````diff
@@ -63,4 +67,8 @@ To learn more about deploying and scaling {{fleet-server}}, refer to:
 
+```{applies_to}
+serverless: unavailable
+```
````
This is the only thing I'm not sure of off the top of my head. If you are sure then 👍

vishaangelova replied:
The reason why I thought this is `serverless: unavailable` is that it references a Fleet Server configuration which is described in [fleet-server-secrets.md](https://www.elastic.co/docs/reference/fleet/fleet-server-secrets). I thought since Fleet Server is managed on Serverless, these configurations are not available there… But now, looking at the content of that file, I think I might have been wrong…

The referenced `fleet-server-secrets.md` doc mentions the following:

> The configuration fragments shown below are specified either in the UI as part of the output specification or as part of the Fleet Server integration settings.

- Fleet Server integration settings” - the Fleet Server integration is not available on Serverless;
- The Fleet UI output configuration - the default output is not editable, but if you add a new ES output, you can edit the SSL fields (but not the host URL):

<img width="587" height="775" alt="Screenshot 2025-12-10 at 10 04 18" src="https://github.com/user-attachments/assets/f6c9760f-40fc-4235-b331-ac202ab4db6d" />


> When you are running Fleet Server under Elastic Agent, you can specify it with either the `--fleet-server-service-token` or the `--fleet-server-service-token-path` flag.
> […] When you are running Fleet Server under Elastic Agent, you can provide the private key path using with the `--fleet-server-cert-key` flag. 
> […] When you are running Fleet Server under Elastic Agent, you can provide the passphrase path using the `--fleet-server-cert-key-passphrase` flag.

On Serverless, I think, you wouldn’t be able to use these for setting up a self-managed Fleet Server, but I’m not sure whether you can set up the service token for the Fleet Server on Serverless when you use that with a proxy server? Maybe yes?

On Serverless, you also cannot configure the default Fleet Server host, but you can add a new one to set it up with a proxy, and then the UI fields are available for that - so these options can still be used for Serverless?

<img width="588" height="761" alt="Screenshot 2025-12-10 at 10 01 05" src="https://github.com/user-attachments/assets/51413c7e-04ee-4040-ba14-b6028c8ab34f" />


This all relates to my [other comment](https://github.com/elastic/docs-content/pull/3238#discussion_r2605663877), and so depending on what is the truth here, those Fleet Server options might still be applicable to Serverless.

vishaangelova replied:
Wait, I’m not entirely sure I get this, but if I understand what you’re saying, then [my suggestion](https://github.com/elastic/docs-content/pull/3238#discussion_r2605663877) to 

> Add a serverless: unavailable tag to those options (for both commands) which have --fleet-server in them. 

might not actually be a good one. Because you’re saying you _can_ use the options on a Fleet Server behind a proxy in the Fleet UI?

vishaangelova replied:
Ahh, now I understand what you mean. Ok, that makes sense.

bmorelli25 commented on reference/fleet/secure-logstash-connections.md:
```diff
@@ -1,6 +1,8 @@
 mapped_pages:
   - https://www.elastic.co/guide/en/fleet/current/secure-logstash-connections.html
+applies_to:
+  stack: ga
```
I think [this page](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/3238/reference/fleet/secure-logstash-connections) is applicable to serverless and other deployment methods.

certutil is called out as being unavailable on cloud, but the logstash output is not unique to self-managed AFAIK

bmorelli25 commented on reference/fleet/migrate-from-beats-to-elastic-agent.md:
```diff
@@ -1,7 +1,11 @@
 ---
 mapped_pages:
```
Should add a serverless: unavailable badge to the heading titled "[Set up Fleet Server (self-managed deployments only)](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/3238/reference/fleet/migrate-from-beats-to-elastic-agent#_set_up_fleet_server_self_managed_deployments_only)" on this page

vishaangelova commented on reference/fleet/fleet-server.md:
```diff
@@ -1,6 +1,8 @@
 mapped_pages:
   - https://www.elastic.co/guide/en/fleet/current/fleet-server.html
+applies_to:
+  stack: ga
```
Yes… I wasn’t sure whether to include it or not at first, but I considered it, and thought I had added it to this file later. Well… no. 🤦 Will add it with the next commit.

vishaangelova commented on reference/fleet/fleet-server.md:
```diff
@@ -3,12 +3,13 @@ mapped_pages:
 ---
 
-# {{fleet-server}} [fleet-server]
+# About {{fleet-server}} [fleet-server]
```
The title of this doc used to be “What is Fleet Server?” but because Vale flagged it, I decided to update it in https://github.com/elastic/docs-content/pull/4245/commits/35edf5692407c67594a46e6294bd981c62382e7b. Well, now I realized there is already a Fleet Server heading in [reference/fleet/index.md](https://github.com/elastic/docs-content/pull/3238/commits/233a90f3ea5a88df7d5ec08cb4093f2e3eda5785#diff-6e1b5715dcb582a948abd3a6e8ab5ec460cd3776724038c2348a4358ee152775R124), which is problematic for SEO, so I decided to change this to “About Fleet Server”. Would that work?

alexandra5000 commented on reference/fleet/dns-processor.md:
```diff
@@ -2,6 +2,9 @@
   - https://www.elastic.co/guide/en/fleet/current/dns-processor.html
+applies_to:
+  stack: ga
+  serverless: ga
```
Not saying this is not true, but this one relies on network access from the processing runtime behavior, and in Serverless outbound network access might be restricted, or proxied, etc.

alexandra5000 commented on reference/fleet/add_docker_metadata-processor.md:
```diff
@@ -2,6 +2,9 @@
   - https://www.elastic.co/guide/en/fleet/current/add_docker_metadata-processor.html
+applies_to:
+  stack: ga
+  serverless: ga
```
Ok, having discussed my doubts with @vishaangelova on Slack, I don't think this is a problem after all, as the events are enriched first, and *then* sent to Serverless. 

The processors look good to me in that case!

karenzone commented on reference/fleet/example-standalone-monitor-nginx-serverless.md:
```diff
@@ -1,6 +1,9 @@
 mapped_pages:
   - https://www.elastic.co/guide/en/fleet/current/example-standalone-monitor-nginx-serverless.html
+applies_to:
+  stack: ga
```
This tutorial is specific to Serverless. Step 2: Create an Elastic Cloud Serverless Project. Even if this could work on Stack or Hosted with a few modifications, I don't think we should expect users to fill in those gaps.
Consider removing stack: ga.

karenzone commented on reference/fleet/hints-annotations-autodiscovery.md:
```diff
@@ -1,6 +1,9 @@
   - https://www.elastic.co/guide/en/fleet/current/hints-annotations-autodiscovery.html
+applies_to:
+  stack: ga
+  serverless: ga
```
Subtopic of Kubernetes autodiscover.
Kubernetes not available on Serverless.

karenzone commented on reference/fleet/migrate-elastic-agent.md:
```diff
@@ -2,6 +2,8 @@
 mapped_pages:
   - https://www.elastic.co/guide/en/fleet/current/migrate-elastic-agent.html
+applies_to:
+  stack: ga
```
Related issue: https://github.com/elastic/docs-content/issues/4239#issuecomment-3627524092

TL;DR: IIUC after similar conversations with Nima, this applies for Serverless, too.

vishaangelova replied:
The problem is that this doc is talking about creating a target cluster from a snapshot. You cannot do that with a Serverless project or at least I couldn’t find a way to do that. Also there’s another step in this doc - [Modify the Fleet Server host](https://www.elastic.co/docs/reference/fleet/migrate-elastic-agent#migrate-elastic-agent-fleet-host) - which cannot be done in Serverless. That said, I should probably add `serverless: unavailable` here, I don’t know why I hadn’t.. maybe I thought the introduction under the heading might be relevant to Serverless, but not the rest of the headings in the doc...

Ultimately, I think most of this doc is not Serverless-compatible at this point (consequently, you might have to revisit https://github.com/elastic/docs-content/issues/4239#issuecomment-3627524092). If there is a way to migrate Elastic Agents to Serverless, we’ll probably have to document it separately.

karenzone replied:
As discussed, we can leave as is, and continue to track and discuss in https://github.com/elastic/docs-content/issues/4239.

karenzone commented on reference/fleet/example-kubernetes-fleet-managed-agent-helm.md:
```diff
@@ -3,7 +3,7 @@ mapped_pages:
   - https://www.elastic.co/guide/en/fleet/current/example-kubernetes-fleet-managed-agent-helm.html
 applies_to:
   stack: ga 8.18
```
Not a line you changed, but indicating that this feature is supported only on `8.18` doesn't seem right. I posted a question in the slack channel.

vishaangelova commented on reference/fleet/example-standalone-monitor-nginx.md:
```diff
@@ -1,6 +1,9 @@
   - https://www.elastic.co/guide/en/fleet/current/example-standalone-monitor-nginx.html
+applies_to:
+  stack: ga
+  serverless: ga
```
Similarly to the other one, I’ll change the `applies_to` here to show the specific deployment the doc is about, rather than “Stack”.

vishaangelova commented on reference/fleet/elastic-agent-kubernetes-autodiscovery.md:
```diff
@@ -1,6 +1,9 @@
   - https://www.elastic.co/guide/en/fleet/current/elastic-agent-kubernetes-autodiscovery.html
+applies_to:
+  stack: ga
+  serverless: ga
```
My understanding here is that we’re talking about Elastic Agent installed in a Kubernetes container, and we’re not talking about it being installed/run in ECK. So I think the described use case is for a host machine on which the user is running a Kubernetes container and Elastic Agent is installed in that container. In this case, they can still send the ingested data from the Elastic Agent running in the Kubernetes container to any Stack deployment, including Serverless. Or do I have that wrong?

vishaangelova commented on reference/fleet/kafka-output.md:
```diff
@@ -2,6 +2,9 @@
   - https://www.elastic.co/guide/en/fleet/current/kafka-output.html
+applies_to:
+  stack: ga
+  serverless: ga
```
Good point… I just realized this is the doc for setting up the Kafka outout for standalone agents, which is different from the doc for setting up the output for Fleet-managed agents (which has the `serverless: ga` tag because you _set up_ the output in the Fleet UI, including on Serverless). But do I understand correctly, that you think we should still keep the stack and serverless tags here?

vishaangelova commented on manage-data/ingest/ingest-reference-architectures.md:
```diff
@@ -19,6 +19,10 @@ To ingest data into {{es}}, use the *simplest option that meets your needs* and
 ::::
 
+::::{admonition} Elastic Cloud Serverless architectural differences
+The architecture diagrams below show {{es}} and {{kib}} as separate components. In {{serverless-full}}, they are combined into a single integrated architecture: {{es-serverless}}.
```
Done in https://github.com/elastic/docs-content/pull/3238/commits/38696ee4d2f53d5c3f74a200f871ba006a03ff11

vishaangelova commented:
@bmorelli25, now that Karen has approved, do you think we can merge this? I've addressed your comments in https://github.com/elastic/docs-content/pull/3238/commits/d0dd04857e5381c29472071e09bcecf5a08b9be0, https://github.com/elastic/docs-content/pull/3238/commits/233a90f3ea5a88df7d5ec08cb4093f2e3eda5785, and https://github.com/elastic/docs-content/pull/3238/commits/796e752a90449189ceda43df81d30aa904fb47e5. Let me know if I missed something and I’ll have a look. If not, feel free to merge!
